src/nti/site/zcml.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+.. $Id$
+"""
+
+from __future__ import print_function, absolute_import, division
+__docformat__ = "restructuredtext en"
+
+logger = __import__('logging').getLogger(__name__)
+
+from zope import interface
+
+from zope.component.zcml import utility
+
+from zope.schema import TextLine
+
+from nti.site._compat import text_
+
+from nti.site.interfaces import ISiteMapping
+
+from nti.site.site import SiteMapping
+
+
+class ISiteMappingDirective(interface.Interface):
+    """
+    Register an :class:`ISiteMapping`
+    """
+    source_site_name = TextLine(title=u"The source site name")
+
+    target_site_name = TextLine(title=u"The target site name")
+
+
+def registerSiteMapping(_context, source_site_name, target_site_name):
+    """
+    Create and register a site mapping, as a utility under the `source_site_name`.
+    """
+    site_mapping = SiteMapping( source_site_name=text_(source_site_name),
```

Comment:
The call to `text_` should not be needed because the schema should have already done that.